test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs

```diff
@@ -7276,6 +7276,142 @@ public void PackCommand_RequireLicenseAcceptanceNotEmitted()
                 }
             }
         }
+
+        [Fact]
+        public void PackCommand_NoRepo_GlobAllFiles_WithDefaultNuspec_Succeeds()
+        {
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var workingDirectory = TestDirectory.Create())
+            {
+                // Arrange
+                Util.CreateFile(
+                    Path.Combine(workingDirectory, "lib", "uap10.0"),
+                    "a.dll",
+                    string.Empty);
+
+                Util.CreateFile(
+                    Path.Combine(workingDirectory, "tools"),
+                    "install.ps1",
+                    string.Empty);
+
+                Util.CreateFile(
+                    Path.Combine(workingDirectory, "images"),
+                    "1.png",
+                    string.Empty);
+
+                Util.CreateFile(
+                    workingDirectory,
+                    "data.txt",
+                    string.Empty);
+
+                // Act
+                CommandRunnerResult specResult = CommandRunner.Run(
+                    nugetexe,
+                    workingDirectory,
+                    "spec",
+                    waitForExit: true);
+                CommandRunnerResult packResult = CommandRunner.Run(
+                    nugetexe,
+                    workingDirectory,
+                    "pack Package.nuspec",
+                    waitForExit: true);
+                Assert.True(0 == specResult.ExitCode, specResult.AllOutput);
+                Assert.True(0 == packResult.ExitCode, packResult.AllOutput);
+
+                // Assert
+                var path = Path.Combine(workingDirectory, "Package.1.0.0.nupkg");
+                var package = new OptimizedZipPackage(path);
```

Comment:
Use PackageArchiveReader instead of OptimizedZipPackage. 

I know many of the other tests use it, but that's because we depend on the old NuGet.Core.dll.